ouroboros-network-framework/src/Ouroboros/Network/Protocol/Handshake/Client.hs

```diff
@@ -52,8 +52,8 @@ handshakeClientPeer codec@VersionDataCodec {encodeData, decodeData}
         -- simultaneous open; 'accept' will choose version (the greatest common
         -- version), and check if we can accept received version data.
         Done TokDone $ case acceptOrRefuse codec acceptVersion versions vMap of
-          Right r      -> Right $ HandshakeNegotiationResult r
-          Left vReason -> Left (HandshakeError vReason)
+          Right (r, vNumber, vData) -> Right $ HandshakeNegotiationResult r vNumber vData
+          Left vReason              -> Left (HandshakeError vReason)
```

Comment:
It seems it would be cleaner if `acceptOrRefuse` returns `Either (HandshakeProtocolError versionNumber) (HandshakeResult r versionNumber versionData)`.  You can leave it as a TODO though.

Reply:
I'm not sure if this makes sense since `handshakeServerPeer` uses `acceptOrRefuse` as well.